src/osd/PG.cc

```diff
@@ -1305,6 +1321,8 @@ void PG::calc_replicated_acting(
   vector<int> *want,
   set<pg_shard_t> *backfill,
   set<pg_shard_t> *acting_backfill,
+  const OSDMapRef osdmap,
+  CephContext *cct,
```

Comment:
if what we need is but `osd_force_auth_primary_missing_objects`, why not passing it in instead?